src/proportional-bar/ProportionalBar.js

```diff
@@ -0,0 +1,132 @@
+  margin: 0;
+  white-space: nowrap;
+`;
+
+// if the legend doesn't wrap it should be roughly this size;
+// the chart height will adjust as necessary
+const INITIAL_METADATA_HEIGHT = 28;
+
+const formatAsPct = format(".0%");
+
+export default function ProportionalBar({ data, title }) {
+  // to both avoid janky chart animations and avoid height overflows
+  // when the legend wraps to multiple lines, we need to measure its height
+  // explicitly to determine the chart height
+  const [metadataHeight, setMetadataHeight] = useState(INITIAL_METADATA_HEIGHT);
+
+  // calculate percentages for display
+  const totalValue = sum(data.map(({ value }) => value));
+  const dataWithPct = data.map((record) => ({
+    ...record,
+    pct: formatAsPct(record.value / totalValue),
+  }));
+
+  const TOOLTIP_PADDING = 3;
+
+  return (
+    <ProportionalBarContainer>
+      <ProportionalBarChartWrapper>
+        <ResponsiveOrdinalFrame
+          data={dataWithPct}
+          // bottom margin leaves room for the title and legend
+          margin={{ top: 0, left: 0, right: 0, bottom: metadataHeight }}
+          oAccessor={() => title}
+          pieceHoverAnnotation
+          projection="horizontal"
+          rAccessor="value"
+          renderKey="label"
+          responsiveHeight
+          responsiveWidth
+          style={(d) => ({ fill: d.color })}
+          tooltipContent={(d) => {
+            return (
+              <Tooltip
+                style={{
+                  // d.y is the vertical center of the hover target; since we know there is only one bar,
+                  // we know it is the vertical center of the entire chart and we can use it
+                  // to push the tooltip below the chart
+                  transform: `translateX(-50%) translateY(${
+                    d.y + TOOLTIP_PADDING
+                  }px)`,
+                }}
+              >
+                <ProportionalBarTooltipText>
+                  {d.label}
+                </ProportionalBarTooltipText>
+                <ProportionalBarTooltipText bold>
+                  {d.value} ({d.pct})
+                </ProportionalBarTooltipText>
+              </Tooltip>
+            );
+          }}
+          type="bar"
+        />
+      </ProportionalBarChartWrapper>
+      <Measure
+        bounds
+        onResize={(contentRect) => setMetadataHeight(contentRect.bounds.height)}
+      >
+        {({ measureRef }) => (
+          <ProportionalBarMetadata ref={measureRef}>
+            <ProportionalBarTitle>{title}</ProportionalBarTitle>
+            <ColorLegend items={data} />
+          </ProportionalBarMetadata>
+        )}
+      </Measure>
+    </ProportionalBarContainer>
+  );
+}
+
+ProportionalBar.propTypes = {
+  data: PropTypes.arrayOf(
+    PropTypes.shape({
+      value: PropTypes.number.isRequired,
+      label: PropTypes.string.isRequired,
+      color: PropTypes.string.isRequired,
+    })
+  ).isRequired,
+  title: PropTypes.string.isRequired,
```

Comment:
Maybe I'm misinterpreting something, but the [issue notes](https://github.com/Recidiviz/public-dashboard/issues/15) that this component should allow for variable `height` and `width` which I was assuming would be properties on this component.

Or is the "proportional" part of this component's name meant to imply that this component will respond to the size of its containing element?

Reply:
i am fulfilling that requirement by having this component fit to both the height and width of its container, rather than requiring that explicit height and width be provided as props ... hoping that will help with responsiveness to screen size among other things